cdap-examples/WikipediaPipeline/src/test/java/co/cask/cdap/examples/wikipedia/WikipediaPipelineAppTest.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  private void testWorkflow(WorkflowManager workflowManager) throws Exception {
+    testWorkflow(workflowManager, null);
+  }
+
+  private void testWorkflow(WorkflowManager workflowManager, @Nullable Integer threshold) throws Exception {
+    if (threshold == null) {
+      workflowManager.start();
+    } else {
+      workflowManager.start(ImmutableMap.of("min.page.threshold", String.valueOf(threshold)));
+    }
+    workflowManager.waitForFinish(5, TimeUnit.MINUTES);
+    String pid = getLatestPid(workflowManager.getHistory());
+    WorkflowTokenNodeDetail tokenAtCondition =
+      workflowManager.getTokenAtNode(pid, WikipediaPipelineWorkflow.EnoughDataToProceed.class.getSimpleName(),
+                                     WorkflowToken.Scope.USER, "result");
+    boolean conditionResult = Boolean.parseBoolean(tokenAtCondition.getTokenDataAtNode().get("result"));
+    if (threshold == null) {
+      Assert.assertFalse(conditionResult);
+      assertWorkflowToken(workflowManager, pid, false);
+    } else {
+      Assert.assertTrue(conditionResult);
+      assertWorkflowToken(workflowManager, pid, true);
+    }
+  }
+
+  @Nullable
+  private String getLatestPid(List<RunRecord> history) {
+    String pid = null;
+    long latestStartTime = 0;
+    for (RunRecord runRecord : history) {
+      // OK to use start ts, since we ensure that the next run begins after the previous run finishes in the test
+      if (runRecord.getStartTs() > latestStartTime) {
+        latestStartTime = runRecord.getStartTs();
+        pid = runRecord.getPid();
+      }
+    }
+    return pid;
+  }
+
+  private void assertWorkflowToken(WorkflowManager workflowManager, String pid,
+                                   boolean continueConditionSucceeded) throws NotFoundException {
+    assertTokenAtPageTitlesMRNode(workflowManager, pid);
+    assertTokenAtRawDataMRNode(workflowManager, pid, continueConditionSucceeded);
+    assertTokenAtNormalizationMRNode(workflowManager, pid, continueConditionSucceeded);
+    assertTokenAtSparkLDANode(workflowManager, pid, continueConditionSucceeded);
+    // TODO: Verification fails if the last node of the workflow is a fork node.
+//    assertTokenAtTopNMRNode(workflowManager, pid, continueConditionSucceeded);
+  }
+
+  private void assertTokenAtPageTitlesMRNode(WorkflowManager workflowManager, String pid) throws NotFoundException {
+    WorkflowTokenNodeDetail pageTitlesUserTokens =
+      workflowManager.getTokenAtNode(pid, WikipediaPipelineApp.LIKES_TO_DATASET_MR_NAME, null, null);
+    Assert.assertTrue(Boolean.parseBoolean(pageTitlesUserTokens.getTokenDataAtNode().get("result")));
+    WorkflowTokenNodeDetail pageTitlesSystemTokens =
+      workflowManager.getTokenAtNode(pid, WikipediaPipelineApp.LIKES_TO_DATASET_MR_NAME,
+                                     WorkflowToken.Scope.SYSTEM, null);
+    Assert.assertEquals(2, Integer.parseInt(pageTitlesSystemTokens.getTokenDataAtNode().get("custom.num.records")));
```

Comment:
The occurrences are across files. I think adding constants in a central class right now for only one Counter may not be worthwhile, plus it makes the code less readable IMO. If we add more counters, maybe we can create an ``enum`` later and use the counter api with enums.